### ISSUE TITLE
[Feature request]: Integrate with Venice AI using their API

### ISSUE THREAD

**Issue by SimonSomlai** [danielmiessler/fabric#610]
### What do you need?

Venice doesn't give API access (yet). But this issue can stay on the backlog until that time ^^. People who use fabric tend to be the privacy-oriented, self-hosting types.

**Comment by timothyjoh** [comment]
Had a look at the communication between the UI and the backend, and it looks pretty simple, but uses Clerk for Auth and such. Would be a good addition when API keys are available, but these models should be available via projects like Ollama as well, right now it is just 

https://huggingface.co/NousResearch/Hermes-2-Theta-Llama-3-8B
and
https://huggingface.co/Dogge/llama-3-70B-uncensored

**Comment by AO12612** [comment]
Api is now available.
Privacy, serious control on system prompt and models that are tendentially less if at all censored are valuable options for serious work.

It is mostly OpenAi compatible, so it should be very simple to integrate (as a matter of fact I have it working in fabric as openai already, but obviously without some finer options/controls)

If devs need api keys the community will be very happy to oblige

https://docs.venice.ai/welcome/about-venice (api related)
https://www.postman.com/veniceai/venice-ai-workspace
https://venice.ai/blog/how-venice-handles-your-privacy
https://discord.gg/askvenice
The community would be happy to share api keys with developers if need be. (or reach out to me)

Roadmap: (implementation requests)
https://featurebase.venice.ai/roadmap/roadmap

**Comment by grasponcrypto** [comment]
Agree, this should be fairly easy to implement and I am, too, am looking forward to integration with venice.ai.  I wrote an integration for home assistant which uses venice.ai and it is very similar to openai api.

**Comment by joe-shenouda** [comment]
Hi All, any progress on this?

**Comment by ksylvan** [comment]
I'll get this into Fabric today @joe-shenouda @grasponcrypto @AO12612 @timothyjoh @SimonSomlai
